cmd/opm/alpha/render/cmd.go

```diff
@@ -0,0 +1,53 @@
+package render
+
+import (
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/operator-framework/operator-registry/internal/action"
+	"github.com/operator-framework/operator-registry/internal/declcfg"
+)
+
+func NewCmd() *cobra.Command {
+	var (
+		render action.Render
+		output string
+	)
+	cmd := &cobra.Command{
+		Use:   "render <index-or-bundle-image1> <index-or-bundle-image2> <index-or-bundle-imageN>",
+		Short: "Generate declarative config blobs from the provided index and bundle images",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			render.Refs = args
+
+			var write func(declcfg.DeclarativeConfig, io.Writer) error
+			switch output {
+			case "yaml":
+				write = declcfg.WriteYAML
+			case "json":
+				write = declcfg.WriteJSON
+			default:
+				log.Fatalf("invalid --output value %q, expected (json|yaml)", output)
+			}
+
+			// TODO(joelanford): What's this for?
+			logrus.SetOutput(ioutil.Discard)
```

Comment:
Similar problem here as in my previous comment. The bundle image parser has some fairly verbose logging in the "happy" case. For example, https://github.com/operator-framework/operator-registry/blob/c95be3c757ffedf26e8ceb89b6d9462ced1604c6/pkg/registry/imageinput.go#L66-L68

Same question: Is it reasonable to expect that any messages that need to be surfaced to the user are returned as errors rather than logged?

Reply:
Similar problem here as in my previous comment. The bundle image parser has some fairly verbose logging in the "happy" case. For example, https://github.com/operator-framework/operator-registry/blob/c95be3c757ffedf26e8ceb89b6d9462ced1604c6/pkg/registry/imageinput.go#L66-L68

Same question: Is it reasonable to expect that any messages that need to be surfaced to the user are returned as errors rather than logged?

Reply:
I think the only case where logs really matter is when debugging. In that case, writing debug logs to stderr works and is idiomatic to my knowledge.

I’m not going to block this PR on logging issues though.

Reply:
Yeah, I'm thinking any logging in these libraries should be at the debug level. That way callers aren't forced into this all or nothing situation.

I'll open a new issue against operator-registry so we don't forget about this.

Reply:
https://github.com/operator-framework/operator-registry/issues/672